### PR TITLE
Add connect to the REST API. Closes #26

### LINKED ISSUE #26: REST API for annotation pipeline
This probably is the best issue to track this, so I am adding it here:

What's lacking in the current architecture (Non-Exhaustive List):
1. Uses Component and Port Names and Annotation Colors from a JSON file rather than requesting server.
2. With #25, port detection workflow has been added but it uses example json file rather than server capabilities. See the workflow checklist in #25, [here](https://github.com/symbench/scm/pull/25#issue-657946863).

For enhancements, we need a a well defined REST API with the following capabilities:
1. Endpoints that get rid of the `json` files from the front end.
2. Annotation Pipeline endpoints with port detection capbilities.

This is just a suggestion, but I think it will get most of the issues resolved.

_Originally posted by @umesh-timalsina in https://github.com/symbench/scm/issues/2#issuecomment-851175991_

### PR DESCRIPTION
**Pull request by brollb**
This should be the last of the required endpoints for the labeling pipeline.

### CODE REVIEW

File: circuit_recognizer/circuit_recognizer/server.py [hunk 1]
```diff
@@ -26,7 +27,7 @@
 )
 
 class ComponentRecognitionRequest(BaseModel):
-    annotations: List[Annotation]
+    annotations: List[Dict[str, Any]]
```
Comment (umesh-timalsina) [inline circuit_recognizer/circuit_recognizer/server.py:30]: If annotations are available can't we just use the `source` to get the image url?

File: circuit_recognizer/circuit_recognizer/server.py [hunk 2]
```diff
@@ -37,18 +38,41 @@ async def detect_ports(annotation: ComponentAnnotation):
...
+    except Exception as e:
+        raise HTTPException(status_code=400, detail=e.args[0])
+
+    get_image_url = lambda ann: ann.target['source']
+    annotations.sort(key=get_image_url)
+    results = []
+    for (image_url, annotations) in groupby(annotations, get_image_url):
+        annotations = list(annotations)
+        image = clean_image(cv2.imread(get_image_path(image_url), cv2.IMREAD_UNCHANGED))
+        edges = connect.collect_edges(image, annotations)
```
Comment (umesh-timalsina) [inline circuit_recognizer/circuit_recognizer/server.py:67]: What kind of annotation is the `collect_edges` method here expecting?

**Review by umesh-timalsina — COMMENTED** [review]
Couple of questions regarding the endpoints?

File: circuit_recognizer/circuit_recognizer/server.py (hunk 1 above)
Comment (brollb) [reply circuit_recognizer/circuit_recognizer/server.py:30]: I was thinking about multi-image scenarios. For example, we load 10 circuits and label a resistor in one of them. We would like to use this annotation on all 10 images (not just the one containing the annotated resistor).

Comment (umesh-timalsina) [reply circuit_recognizer/circuit_recognizer/server.py:30]: Got it.

File: circuit_recognizer/circuit_recognizer/server.py (hunk 2 above)
Comment (umesh-timalsina) [reply circuit_recognizer/circuit_recognizer/server.py:67]: By full we mean not a single component and its ports should be missing from the annotations right?

**Pull request merged**